builtin/credential/aws/path_login_test.go

```diff
@@ -214,8 +214,38 @@ func TestBackend_pathLogin_IAMHeaders(t *testing.T) {
 		t.Fatal(err)
 	}
 
+	// Configure identity.
+	_, err = b.HandleRequest(context.Background(), &logical.Request{
+		Operation: logical.UpdateOperation,
+		Path:      "config/identity",
+		Storage:   storage,
+		Data: map[string]interface{}{
+			"iam_alias": "role_id",
+			"iam_metadata": []string{
+				"default",
+				"canonical_arn",
+				"client_arn",
+				"client_user_id",
+				"inferred_aws_region",
+				"inferred_entity_id",
+				"inferred_entity_type",
+			},
+			"ec2_alias": "role_id",
+			"ec2_metadata": []string{
+				"default",
+				"ami_id",
+				"instance_id",
+				"region",
+			},
+		},
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
```

Comment:
Do we also want to add a test that tests the default case?

Reply:
Added this here: https://github.com/hashicorp/vault/pull/8783/files#diff-a3499e6e02611fff417abe5da926b5fdR379

Looks like tests are now broken remotely, so here's the local test output:

<details>
<summary>AWS dir</summary>

```
tbex@pop-os:~/go/src/github.com/hashicorp/vault/builtin/credential/aws$ go test -v
=== RUN   TestBackend_E2E_Initialize
2020-04-22T11:06:38.199-0700 [DEBUG] storage.cache: creating LRU cache: size=0
2020-04-22T11:06:38.200-0700 [ERROR] core: no seal config found, can't determine if legacy or new-style shamir
2020-04-22T11:06:38.200-0700 [ERROR] core: no seal config found, can't determine if legacy or new-style shamir
2020-04-22T11:06:38.200-0700 [ERROR] core: no seal config found, can't determine if legacy or new-style shamir
2020-04-22T11:06:38.200-0700 [INFO]  core: security barrier not initialized
2020-04-22T11:06:38.200-0700 [INFO]  core: security barrier initialized: stored=1 shares=3 threshold=3
2020-04-22T11:06:38.200-0700 [DEBUG] core: cluster name not found/set, generating new
2020-04-22T11:06:38.200-0700 [DEBUG] core: cluster name set: name=vault-cluster-53d87472
2020-04-22T11:06:38.200-0700 [DEBUG] core: cluster ID not found, generating new
2020-04-22T11:06:38.200-0700 [DEBUG] core: cluster ID set: id=844531c9-2e4c-f5e0-d97d-b436ec4e0652
2020-04-22T11:06:38.200-0700 [DEBUG] core: generating cluster private key
2020-04-22T11:06:38.209-0700 [DEBUG] core: generating local cluster certificate
2020-04-22T11:06:38.218-0700 [INFO]  core: post-unseal setup starting
2020-04-22T11:06:38.218-0700 [DEBUG] core: clearing forwarding clients
2020-04-22T11:06:38.218-0700 [DEBUG] core: done clearing forwarding clients
2020-04-22T11:06:38.218-0700 [DEBUG] core: persisting feature flags
2020-04-22T11:06:38.228-0700 [INFO]  core: loaded wrapping token key
2020-04-22T11:06:38.228-0700 [INFO]  core: successfully setup plugin catalog: plugin-directory=
2020-04-22T11:06:38.228-0700 [INFO]  core: no mounts; adding default mount table
2020-04-22T11:06:38.229-0700 [INFO]  core: successfully mounted backend: type=cubbyhole path=cubbyhole/
2020-04-22T11:06:38.229-0700 [INFO]  core: successfully mounted backend: type=system path=sys/
2020-04-22T11:06:38.229-0700 [INFO]  core: successfully mounted backend: type=identity path=identity/
2020-04-22T11:06:38.231-0700 [INFO]  core: successfully enabled credential backend: type=token path=token/
2020-04-22T11:06:38.231-0700 [INFO]  core: restoring leases
2020-04-22T11:06:38.231-0700 [INFO]  rollback: starting rollback manager
2020-04-22T11:06:38.231-0700 [DEBUG] expiration: collecting leases
2020-04-22T11:06:38.231-0700 [DEBUG] expiration: leases collected: num_existing=0
2020-04-22T11:06:38.231-0700 [INFO]  expiration: lease restore complete
2020-04-22T11:06:38.232-0700 [DEBUG] identity: loading entities
2020-04-22T11:06:38.232-0700 [DEBUG] identity: entities collected: num_existing=0
2020-04-22T11:06:38.232-0700 [INFO]  identity: entities restored
2020-04-22T11:06:38.233-0700 [DEBUG] identity: identity loading groups
2020-04-22T11:06:38.233-0700 [DEBUG] identity: groups collected: num_existing=0
2020-04-22T11:06:38.233-0700 [INFO]  identity: groups restored
2020-04-22T11:06:38.233-0700 [INFO]  core: post-unseal setup complete
2020-04-22T11:06:38.233-0700 [INFO]  core: root token generated
2020-04-22T11:06:38.233-0700 [INFO]  core: pre-seal teardown starting
2020-04-22T11:06:38.233-0700 [DEBUG] expiration: stop triggered
2020-04-22T11:06:38.233-0700 [DEBUG] expiration: finished stopping
2020-04-22T11:06:38.233-0700 [INFO]  rollback: stopping rollback manager
2020-04-22T11:06:38.233-0700 [INFO]  core: pre-seal teardown complete
2020-04-22T11:06:38.233-0700 [DEBUG] core: unseal key supplied
2020-04-22T11:06:38.233-0700 [DEBUG] core: cannot unseal, not enough keys: keys=1 threshold=3 nonce=90158d41-dd2a-f88f-f512-46a8a7e0d44b
2020-04-22T11:06:38.233-0700 [DEBUG] core: unseal key supplied
2020-04-22T11:06:38.233-0700 [DEBUG] core: cannot unseal, not enough keys: keys=2 threshold=3 nonce=90158d41-dd2a-f88f-f512-46a8a7e0d44b
2020-04-22T11:06:38.233-0700 [DEBUG] core: unseal key supplied
2020-04-22T11:06:38.233-0700 [DEBUG] core: starting cluster listeners
2020-04-22T11:06:38.233-0700 [INFO]  core.cluster-listener.tcp: starting listener: listener_address=127.0.0.1:0
2020-04-22T11:06:38.233-0700 [INFO]  core.cluster-listener: serving cluster requests: cluster_listen_address=127.0.0.1:38723
2020-04-22T11:06:38.233-0700 [INFO]  core: vault is unsealed
2020-04-22T11:06:38.233-0700 [INFO]  core: entering standby mode
2020-04-22T11:06:38.233-0700 [INFO]  core: acquired lock, enabling active operation
2020-04-22T11:06:38.233-0700 [DEBUG] core: generating cluster private key
2020-04-22T11:06:38.242-0700 [DEBUG] core: generating local cluster certificate
2020-04-22T11:06:38.251-0700 [INFO]  core: post-unseal setup starting
2020-04-22T11:06:38.251-0700 [DEBUG] core: clearing forwarding clients
2020-04-22T11:06:38.251-0700 [DEBUG] core: done clearing forwarding clients
2020-04-22T11:06:38.251-0700 [DEBUG] core: persisting feature flags
2020-04-22T11:06:38.251-0700 [INFO]  core: loaded wrapping token key
2020-04-22T11:06:38.251-0700 [INFO]  core: successfully setup plugin catalog: plugin-directory=
2020-04-22T11:06:38.251-0700 [INFO]  core: successfully mounted backend: type=system path=sys/
2020-04-22T11:06:38.251-0700 [INFO]  core: successfully mounted backend: type=identity path=identity/
2020-04-22T11:06:38.251-0700 [INFO]  core: successfully mounted backend: type=cubbyhole path=cubbyhole/
2020-04-22T11:06:38.253-0700 [INFO]  core: successfully enabled credential backend: type=token path=token/
2020-04-22T11:06:38.253-0700 [INFO]  core: restoring leases
2020-04-22T11:06:38.253-0700 [DEBUG] identity: loading entities
2020-04-22T11:06:38.253-0700 [DEBUG] identity: entities collected: num_existing=0
2020-04-22T11:06:38.253-0700 [INFO]  rollback: starting rollback manager
2020-04-22T11:06:38.253-0700 [DEBUG] expiration: collecting leases
2020-04-22T11:06:38.253-0700 [DEBUG] expiration: leases collected: num_existing=0
2020-04-22T11:06:38.253-0700 [INFO]  identity: entities restored
2020-04-22T11:06:38.253-0700 [DEBUG] identity: identity loading groups
2020-04-22T11:06:38.253-0700 [DEBUG] identity: groups collected: num_existing=0
2020-04-22T11:06:38.253-0700 [INFO]  identity: groups restored
2020-04-22T11:06:38.253-0700 [DEBUG] core: request forwarding setup function
2020-04-22T11:06:38.253-0700 [DEBUG] core: clearing forwarding clients
2020-04-22T11:06:38.253-0700 [DEBUG] core: done clearing forwarding clients
2020-04-22T11:06:38.253-0700 [DEBUG] core: leaving request forwarding setup function
2020-04-22T11:06:38.253-0700 [INFO]  core: post-unseal setup complete
2020-04-22T11:06:38.253-0700 [INFO]  expiration: lease restore complete
2020-04-22T11:06:38.254-0700 [INFO]  core: successful mount: namespace= path=secret/ type=kv
2020-04-22T11:06:38.255-0700 [INFO]  core: enabled audit backend: path=noop/ type=noop
2020-04-22T11:06:38.258-0700 [DEBUG] auth.aws.auth_aws_a9ae3d79.initialize: starting initialization
2020-04-22T11:06:38.258-0700 [INFO]  core: enabled credential backend: path=aws/ type=aws
2020-04-22T11:06:38.258-0700 [INFO]  auth.aws.auth_aws_a9ae3d79.initialize: an upgrade was performed during initialization
2020-04-22T11:06:39.267-0700 [INFO]  core: marked as sealed
2020-04-22T11:06:39.267-0700 [DEBUG] core: clearing forwarding clients
2020-04-22T11:06:39.267-0700 [DEBUG] core: done clearing forwarding clients
2020-04-22T11:06:39.267-0700 [DEBUG] core: finished triggering standbyStopCh for runStandby
2020-04-22T11:06:39.267-0700 [DEBUG] core: shutting down periodic key rotation checker
2020-04-22T11:06:39.267-0700 [DEBUG] core: shutting down periodic leader refresh
2020-04-22T11:06:39.268-0700 [DEBUG] core: shutting down leader elections
2020-04-22T11:06:39.268-0700 [INFO]  core: pre-seal teardown starting
2020-04-22T11:06:39.768-0700 [DEBUG] expiration: stop triggered
2020-04-22T11:06:39.768-0700 [DEBUG] expiration: finished stopping
2020-04-22T11:06:39.768-0700 [INFO]  rollback: stopping rollback manager
2020-04-22T11:06:39.768-0700 [INFO]  core: pre-seal teardown complete
2020-04-22T11:06:39.768-0700 [DEBUG] core: runStandby done
2020-04-22T11:06:39.768-0700 [INFO]  core: stopping cluster listeners
2020-04-22T11:06:39.768-0700 [INFO]  core.cluster-listener: forwarding rpc listeners stopped
2020-04-22T11:06:40.234-0700 [INFO]  core.cluster-listener: rpc listeners successfully shut down
2020-04-22T11:06:40.234-0700 [INFO]  core: cluster listeners successfully shut down
2020-04-22T11:06:40.234-0700 [DEBUG] core: sealing barrier
2020-04-22T11:06:40.234-0700 [INFO]  core: vault is sealed
2020-04-22T11:06:40.234-0700 [DEBUG] core: unseal key supplied
2020-04-22T11:06:40.234-0700 [DEBUG] core: cannot unseal, not enough keys: keys=1 threshold=3 nonce=ff0401cb-502e-dc9d-2255-50ba3e616d4c
2020-04-22T11:06:40.234-0700 [DEBUG] core: unseal key supplied
2020-04-22T11:06:40.234-0700 [DEBUG] core: cannot unseal, not enough keys: keys=2 threshold=3 nonce=ff0401cb-502e-dc9d-2255-50ba3e616d4c
2020-04-22T11:06:40.234-0700 [DEBUG] core: unseal key supplied
2020-04-22T11:06:40.234-0700 [DEBUG] core: starting cluster listeners
2020-04-22T11:06:40.234-0700 [INFO]  core.cluster-listener.tcp: starting listener: listener_address=127.0.0.1:38723
2020-04-22T11:06:40.234-0700 [INFO]  core.cluster-listener: serving cluster requests: cluster_listen_address=127.0.0.1:38723
2020-04-22T11:06:40.234-0700 [INFO]  core: vault is unsealed
2020-04-22T11:06:40.234-0700 [INFO]  core: entering standby mode
2020-04-22T11:06:40.235-0700 [INFO]  core: acquired lock, enabling active operation
2020-04-22T11:06:40.235-0700 [DEBUG] core: generating cluster private key
2020-04-22T11:06:40.246-0700 [DEBUG] core: generating local cluster certificate
2020-04-22T11:06:40.255-0700 [INFO]  core: post-unseal setup starting
2020-04-22T11:06:40.255-0700 [DEBUG] core: clearing forwarding clients
2020-04-22T11:06:40.255-0700 [DEBUG] core: done clearing forwarding clients
2020-04-22T11:06:40.255-0700 [DEBUG] core: persisting feature flags
2020-04-22T11:06:40.255-0700 [INFO]  core: loaded wrapping token key
2020-04-22T11:06:40.255-0700 [INFO]  core: successfully setup plugin catalog: plugin-directory=
2020-04-22T11:06:40.256-0700 [INFO]  core: successfully mounted backend: type=system path=sys/
2020-04-22T11:06:40.256-0700 [INFO]  core: successfully mounted backend: type=identity path=identity/
2020-04-22T11:06:40.256-0700 [INFO]  core: successfully mounted backend: type=kv path=secret/
2020-04-22T11:06:40.256-0700 [INFO]  core: successfully mounted backend: type=cubbyhole path=cubbyhole/
2020-04-22T11:06:40.257-0700 [INFO]  core: successfully enabled credential backend: type=token path=token/
2020-04-22T11:06:40.257-0700 [INFO]  core: successfully enabled credential backend: type=aws path=aws/
2020-04-22T11:06:40.257-0700 [INFO]  core: restoring leases
2020-04-22T11:06:40.257-0700 [INFO]  rollback: starting rollback manager
2020-04-22T11:06:40.257-0700 [DEBUG] identity: loading entities
2020-04-22T11:06:40.257-0700 [DEBUG] identity: entities collected: num_existing=0
2020-04-22T11:06:40.257-0700 [DEBUG] expiration: collecting leases
2020-04-22T11:06:40.257-0700 [DEBUG] expiration: leases collected: num_existing=0
2020-04-22T11:06:40.257-0700 [INFO]  identity: entities restored
2020-04-22T11:06:40.257-0700 [DEBUG] identity: identity loading groups
2020-04-22T11:06:40.257-0700 [DEBUG] identity: groups collected: num_existing=0
2020-04-22T11:06:40.257-0700 [INFO]  identity: groups restored
2020-04-22T11:06:40.257-0700 [INFO]  expiration: lease restore complete
2020-04-22T11:06:40.257-0700 [DEBUG] core: request forwarding setup function
2020-04-22T11:06:40.257-0700 [DEBUG] core: clearing forwarding clients
2020-04-22T11:06:40.257-0700 [DEBUG] core: done clearing forwarding clients
2020-04-22T11:06:40.257-0700 [DEBUG] core: leaving request forwarding setup function
2020-04-22T11:06:40.257-0700 [DEBUG] auth.aws.auth_aws_a9ae3d79.initialize: starting initialization
2020-04-22T11:06:40.257-0700 [INFO]  core: post-unseal setup complete
2020-04-22T11:06:40.257-0700 [INFO]  auth.aws.auth_aws_a9ae3d79.initialize: an upgrade was performed during initialization
2020-04-22T11:06:43.257-0700 [INFO]  TestBackend_E2E_Initialize: cleaning up vault cluster
--- PASS: TestBackend_E2E_Initialize (7.05s)
=== RUN   TestBackend_CreateParseVerifyRoleTag
--- PASS: TestBackend_CreateParseVerifyRoleTag (0.00s)
=== RUN   TestBackend_prepareRoleTagPlaintextValue
--- PASS: TestBackend_prepareRoleTagPlaintextValue (0.00s)
=== RUN   TestBackend_CreateRoleTagNonce
--- PASS: TestBackend_CreateRoleTagNonce (0.00s)
=== RUN   TestBackend_ConfigTidyIdentities
--- PASS: TestBackend_ConfigTidyIdentities (0.00s)
=== RUN   TestBackend_ConfigTidyRoleTags
--- PASS: TestBackend_ConfigTidyRoleTags (0.00s)
=== RUN   TestBackend_TidyIdentities
--- PASS: TestBackend_TidyIdentities (1.00s)
=== RUN   TestBackend_TidyRoleTags
--- PASS: TestBackend_TidyRoleTags (1.00s)
=== RUN   TestBackend_ConfigClient
2020-04-22T11:06:47.242-0700 [DEBUG] storage.cache: creating LRU cache: size=0
2020-04-22T11:06:47.242-0700 [ERROR] core: no seal config found, can't determine if legacy or new-style shamir
2020-04-22T11:06:47.242-0700 [INFO]  core: security barrier not initialized
2020-04-22T11:06:47.242-0700 [INFO]  core: security barrier initialized: stored=1 shares=1 threshold=1
2020-04-22T11:06:47.242-0700 [DEBUG] core: cluster name not found/set, generating new
2020-04-22T11:06:47.242-0700 [DEBUG] core: cluster name set: name=vault-cluster-395ba019
2020-04-22T11:06:47.242-0700 [DEBUG] core: cluster ID not found, generating new
2020-04-22T11:06:47.242-0700 [DEBUG] core: cluster ID set: id=82842e9f-50ea-ec5c-dafa-a4c128fbc8e1
2020-04-22T11:06:47.242-0700 [INFO]  core: post-unseal setup starting
2020-04-22T11:06:47.242-0700 [DEBUG] core: clearing forwarding clients
2020-04-22T11:06:47.242-0700 [DEBUG] core: done clearing forwarding clients
2020-04-22T11:06:47.242-0700 [DEBUG] core: persisting feature flags
2020-04-22T11:06:47.253-0700 [INFO]  core: loaded wrapping token key
2020-04-22T11:06:47.253-0700 [INFO]  core: successfully setup plugin catalog: plugin-directory=
2020-04-22T11:06:47.253-0700 [INFO]  core: no mounts; adding default mount table
2020-04-22T11:06:47.254-0700 [INFO]  core: successfully mounted backend: type=cubbyhole path=cubbyhole/
2020-04-22T11:06:47.254-0700 [INFO]  core: successfully mounted backend: type=system path=sys/
2020-04-22T11:06:47.255-0700 [INFO]  core: successfully mounted backend: type=identity path=identity/
2020-04-22T11:06:47.256-0700 [INFO]  core: successfully enabled credential backend: type=token path=token/
2020-04-22T11:06:47.256-0700 [INFO]  core: restoring leases
2020-04-22T11:06:47.256-0700 [INFO]  rollback: starting rollback manager
2020-04-22T11:06:47.256-0700 [DEBUG] expiration: collecting leases
2020-04-22T11:06:47.256-0700 [DEBUG] expiration: leases collected: num_existing=0
2020-04-22T11:06:47.256-0700 [INFO]  expiration: lease restore complete
2020-04-22T11:06:47.257-0700 [DEBUG] identity: loading entities
2020-04-22T11:06:47.257-0700 [DEBUG] identity: entities collected: num_existing=0
2020-04-22T11:06:47.257-0700 [INFO]  identity: entities restored
2020-04-22T11:06:47.257-0700 [DEBUG] identity: identity loading groups
2020-04-22T11:06:47.257-0700 [DEBUG] identity: groups collected: num_existing=0
2020-04-22T11:06:47.257-0700 [INFO]  identity: groups restored
2020-04-22T11:06:47.257-0700 [INFO]  core: post-unseal setup complete
2020-04-22T11:06:47.257-0700 [INFO]  core: root token generated
2020-04-22T11:06:47.257-0700 [INFO]  core: pre-seal teardown starting
2020-04-22T11:06:47.257-0700 [DEBUG] expiration: stop triggered
2020-04-22T11:06:47.257-0700 [DEBUG] expiration: finished stopping
2020-04-22T11:06:47.257-0700 [INFO]  rollback: stopping rollback manager
2020-04-22T11:06:47.257-0700 [INFO]  core: pre-seal teardown complete
2020-04-22T11:06:47.257-0700 [DEBUG] core: unseal key supplied
2020-04-22T11:06:47.257-0700 [INFO]  core: clustering disabled, not starting listeners
2020-04-22T11:06:47.257-0700 [INFO]  core: post-unseal setup starting
2020-04-22T11:06:47.257-0700 [DEBUG] core: clearing forwarding clients
2020-04-22T11:06:47.257-0700 [DEBUG] core: done clearing forwarding clients
2020-04-22T11:06:47.257-0700 [DEBUG] core: persisting feature flags
2020-04-22T11:06:47.257-0700 [INFO]  core: loaded wrapping token key
2020-04-22T11:06:47.257-0700 [INFO]  core: successfully setup plugin catalog: plugin-directory=
2020-04-22T11:06:47.257-0700 [INFO]  core: successfully mounted backend: type=system path=sys/
2020-04-22T11:06:47.257-0700 [INFO]  core: successfully mounted backend: type=identity path=identity/
2020-04-22T11:06:47.257-0700 [INFO]  core: successfully mounted backend: type=cubbyhole path=cubbyhole/
2020-04-22T11:06:47.258-0700 [INFO]  core: successfully enabled credential backend: type=token path=token/
2020-04-22T11:06:47.258-0700 [INFO]  core: restoring leases
2020-04-22T11:06:47.258-0700 [DEBUG] identity: loading entities
2020-04-22T11:06:47.258-0700 [DEBUG] identity: entities collected: num_existing=0
2020-04-22T11:06:47.258-0700 [INFO]  rollback: starting rollback manager
2020-04-22T11:06:47.258-0700 [INFO]  identity: entities restored
2020-04-22T11:06:47.258-0700 [DEBUG] identity: identity loading groups
2020-04-22T11:06:47.258-0700 [DEBUG] expiration: collecting leases
2020-04-22T11:06:47.258-0700 [DEBUG] expiration: leases collected: num_existing=0
2020-04-22T11:06:47.258-0700 [DEBUG] identity: groups collected: num_existing=0
2020-04-22T11:06:47.258-0700 [INFO]  identity: groups restored
2020-04-22T11:06:47.258-0700 [INFO]  core: post-unseal setup complete
2020-04-22T11:06:47.258-0700 [INFO]  core: vault is unsealed
2020-04-22T11:06:47.258-0700 [INFO]  expiration: lease restore complete
2020-04-22T11:06:47.259-0700 [DEBUG] initialize: starting initialization
2020-04-22T11:06:47.259-0700 [INFO]  core: enabled credential backend: path=mnt/ type=test
2020-04-22T11:06:47.259-0700 [INFO]  initialize: an upgrade was performed during initialization
2020-04-22T11:06:47.260-0700 [WARN]  Executing test step: step_number=1
2020-04-22T11:06:47.261-0700 [WARN]  Executing test step: step_number=2
2020-04-22T11:06:47.261-0700 [WARN]  Executing test step: step_number=3
2020-04-22T11:06:47.261-0700 [WARN]  Executing test step: step_number=4
2020-04-22T11:06:47.261-0700 [WARN]  Requesting RollbackOperation
--- PASS: TestBackend_ConfigClient (0.02s)
=== RUN   TestBackend_pathConfigCertificate
--- PASS: TestBackend_pathConfigCertificate (0.00s)
=== RUN   TestBackend_parseAndVerifyRoleTagValue
--- PASS: TestBackend_parseAndVerifyRoleTagValue (0.00s)
=== RUN   TestBackend_PathRoleTag
--- PASS: TestBackend_PathRoleTag (0.00s)
=== RUN   TestBackend_PathBlacklistRoleTag
--- PASS: TestBackend_PathBlacklistRoleTag (0.00s)
=== RUN   TestBackendAcc_LoginWithInstanceIdentityDocAndWhitelistIdentity
--- SKIP: TestBackendAcc_LoginWithInstanceIdentityDocAndWhitelistIdentity (0.00s)
    backend_test.go:1037: Acceptance tests skipped unless env 'VAULT_ACC' set
=== RUN   TestBackend_pathStsConfig
--- PASS: TestBackend_pathStsConfig (0.00s)
=== RUN   TestBackendAcc_LoginWithCallerIdentity
--- SKIP: TestBackendAcc_LoginWithCallerIdentity (0.00s)
    backend_test.go:1447: Acceptance tests skipped unless env 'VAULT_ACC' set
=== RUN   TestGeneratePartitionToRegionMap
--- PASS: TestGeneratePartitionToRegionMap (0.00s)
=== RUN   TestBackend_pathConfigClient
--- PASS: TestBackend_pathConfigClient (0.00s)
=== RUN   TestBackend_pathConfigIdentity
--- PASS: TestBackend_pathConfigIdentity (0.00s)
=== RUN   TestBackend_pathLogin_getCallerIdentityResponse
--- PASS: TestBackend_pathLogin_getCallerIdentityResponse (0.00s)
=== RUN   TestBackend_pathLogin_parseIamArn
--- PASS: TestBackend_pathLogin_parseIamArn (0.00s)
=== RUN   TestBackend_validateVaultHeaderValue
--- PASS: TestBackend_validateVaultHeaderValue (0.00s)
=== RUN   TestBackend_pathLogin_IAMHeaders
=== RUN   TestBackend_pathLogin_IAMHeaders/Default
=== RUN   TestBackend_pathLogin_IAMHeaders/Map-complete
=== RUN   TestBackend_pathLogin_IAMHeaders/Map-incomplete
=== RUN   TestBackend_pathLogin_IAMHeaders/JSON-complete
=== RUN   TestBackend_pathLogin_IAMHeaders/JSON-incomplete
=== RUN   TestBackend_pathLogin_IAMHeaders/Base64-complete
=== RUN   TestBackend_pathLogin_IAMHeaders/Base64-incomplete-missing-header
=== RUN   TestBackend_pathLogin_IAMHeaders/Base64-incomplete-missing-auth-sig
--- PASS: TestBackend_pathLogin_IAMHeaders (0.00s)
    --- PASS: TestBackend_pathLogin_IAMHeaders/Default (0.00s)
    --- PASS: TestBackend_pathLogin_IAMHeaders/Map-complete (0.00s)
    --- PASS: TestBackend_pathLogin_IAMHeaders/Map-incomplete (0.00s)
    --- PASS: TestBackend_pathLogin_IAMHeaders/JSON-complete (0.00s)
    --- PASS: TestBackend_pathLogin_IAMHeaders/JSON-incomplete (0.00s)
    --- PASS: TestBackend_pathLogin_IAMHeaders/Base64-complete (0.00s)
    --- PASS: TestBackend_pathLogin_IAMHeaders/Base64-incomplete-missing-header (0.00s)
    --- PASS: TestBackend_pathLogin_IAMHeaders/Base64-incomplete-missing-auth-sig (0.00s)
=== RUN   TestBackend_defaultAliasMetadata
=== RUN   TestBackend_defaultAliasMetadata/Default
=== RUN   TestBackend_defaultAliasMetadata/Map-complete
=== RUN   TestBackend_defaultAliasMetadata/JSON-complete
=== RUN   TestBackend_defaultAliasMetadata/Base64-complete
--- PASS: TestBackend_defaultAliasMetadata (0.00s)
    --- PASS: TestBackend_defaultAliasMetadata/Default (0.00s)
    --- PASS: TestBackend_defaultAliasMetadata/Map-complete (0.00s)
    --- PASS: TestBackend_defaultAliasMetadata/JSON-complete (0.00s)
    --- PASS: TestBackend_defaultAliasMetadata/Base64-complete (0.00s)
=== RUN   TestBackend_pathRoleEc2
--- PASS: TestBackend_pathRoleEc2 (0.00s)
=== RUN   Test_enableIamIDResolution
--- PASS: Test_enableIamIDResolution (0.00s)
=== RUN   TestBackend_pathIam
--- PASS: TestBackend_pathIam (0.00s)
=== RUN   TestBackend_pathRoleMixedTypes
2020-04-22T11:06:47.274-0700 [DEBUG] no stsRole found for 123456789012
2020-04-22T11:06:47.274-0700 [DEBUG] no cached client for region us-east-1 and stsRole 
--- PASS: TestBackend_pathRoleMixedTypes (0.43s)
=== RUN   TestAwsEc2_RoleCrud
--- PASS: TestAwsEc2_RoleCrud (0.00s)
=== RUN   TestAwsEc2_RoleDurationSeconds
--- PASS: TestAwsEc2_RoleDurationSeconds (0.00s)
=== RUN   TestRoleEntryUpgradeV
--- PASS: TestRoleEntryUpgradeV (0.00s)
=== RUN   TestRoleInitialize
--- PASS: TestRoleInitialize (0.00s)
=== RUN   TestAwsVersion
--- PASS: TestAwsVersion (0.00s)
=== RUN   TestRoleResolutionWithSTSEndpointConfigured
--- SKIP: TestRoleResolutionWithSTSEndpointConfigured (0.00s)
    path_role_test.go:999: 
PASS
ok  	github.com/hashicorp/vault/builtin/credential/aws	9.528s
```

</details>

<details>
<summary>SDK dir</summary>

```
tbex@pop-os:~/go/src/github.com/hashicorp/vault/sdk/helper/aliasmetadata$ go test -v
=== RUN   TestAcceptance
--- PASS: TestAcceptance (0.00s)
=== RUN   TestFieldSchema
--- PASS: TestFieldSchema (0.00s)
=== RUN   TestGetAliasMetadata
--- PASS: TestGetAliasMetadata (0.00s)
=== RUN   TestParseAliasMetadata
--- PASS: TestParseAliasMetadata (0.00s)
=== RUN   TestPopulateDesiredAliasMetadata
--- PASS: TestPopulateDesiredAliasMetadata (0.00s)
PASS
ok  	github.com/hashicorp/vault/sdk/helper/aliasmetadata	0.004s
```

</details>